src/controller/handler/router.go

```diff
@@ -73,6 +76,9 @@ var routes = []Route{
     Route{"GET",   "/api/v1/repos/{namespace}/{name}",            wrapper(authWrapper(globalRepoWrapper(GetGlobalRepo)))},
     Route{"GET",   "/api/v1/repos/{namespace}/{name}/tags",       wrapper(authWrapper(globalRepoWrapper(ListGlobalRepoTag)))},
     Route{"GET",   "/api/v1/repos/{namespace}/{name}/tags/{tag}", wrapper(authWrapper(globalRepoWrapper(GetGlobalRepoTag)))},
+
+    // agent
+    Route{"GET", "/api/v1/agent/{user_key}", wrapperWS(ConnectAgent)},
```

Comment:
From my opinion, we need a pair <key, uuid> to identify a node;
it may be better to store key and uuid in request payload or request header, and formatting api like "/api/v1/ws"